nuxeo-core/nuxeo-core-storage-mem/src/main/java/org/nuxeo/ecm/core/storage/mem/MemRepository.java

```diff
@@ -330,6 +331,27 @@ public boolean queryKeyValuePresence(String key, String value, Set<String> ignor
         return false;
     }
 
+    @Override
+    public List<State> queryKeyValueWithOperator(String key1, Object value1, String key2, DBSQueryOperator operator,
+            Object value2, Set<String> ignored) {
+        List<State> results = new ArrayList<>();
+        switch (operator) {
+        case IN:
+            HashSet<String> ids = new HashSet<>((Collection<? extends String>) value2);
+            states.forEach((id, state) -> {
+                if (Boolean.valueOf((String) state.get(DBSDocument.KEY_IS_VERSION))) {
+                    if (ids.contains(state.get(DBSDocument.KEY_VERSION_SERIES_ID))) {
+                        results.add(state);
+                    }
+                }
```

Comment:
I assume this commit is still a draft because it's not the correct generic implementation for this method.
(I suggest, even for draft code, if it's pushed to a PR, to always add comments like `// XXX testing` or whatever best describes what is being done)